core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1378,20 +1416,30 @@ class UnifiedLog(@volatile var logStartOffset: Long,
     }
   }
 
+  private[log] def localRetentionMs: Long = {
```

Comment:
I would suggest to have address local retention in a separate PR. We can limit this PR to handling remote log retention only.